### ISSUE TITLE
Ampache api with nextcloud served by nginx gets 404 when following nextclou admin docs 

### ISSUE THREAD

**Issue by aleixq** [owncloud/music#851]
Setting nginx site as it's explained in nextcloud administration docs for nginx: https://docs.nextcloud.com/server/21/admin_manual/installation/nginx.html ,  the php location block breaks ampache api as it splits the path from last ".php" found  (xml.server.php) using `fastcgi_split_path_info ^(.+?\.php)(/.*)$` . 
Using https://example.org/index.php/apps/music/ampache/server/xml.server.php works...

**Comment by paulijar** [comment]
Thanks for the info. Do you have some suggestion, what should be changed in the Music app? Or is the point that the Nextcloud admin docs should not be followed?

**Comment by aleixq** [comment]
Paul, as ampache api endpoint of music app follows the specification of ampache docs, I think the only way is to add a new location block or tweaking the php handler block in nginx conf. I am gonna test later which one helps there.

**Comment by aleixq** [comment]
It works for me: 
```
--- nextcloud-old.conf  2021-05-05 01:02:35.871870022 +0200
+++ nextcloud.conf      2021-05-05 01:03:11.491904219 +0200
@@ -113,7 +113,7 @@
         fastcgi_split_path_info ^(.+?\.php)(/.*)$;
         set $path_info $fastcgi_path_info;
 
-        try_files $fastcgi_script_name =404;
+        try_files $fastcgi_script_name /index.php$fastcgi_script_name?$args;
 
         include fastcgi_params;
         fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;

```

But as it touches the main php handler of the whole nextcloud maybe there are some problems somewhere ...

**Comment by pled** [comment]
HI,
Same issue here using Docker Nextcloud 29 + Swag containers (Swag is nginx+fail2ban+letencrypt, from linuxserver.io).

I can see this issue open for 3 years, does that mean it will not be fixed ?

**Comment by paulijar** [comment]
@pled I don't believe that there's anything that the Music app could do to help the situation if the problem is in the nginx configuration. Frankly, configuring web servers has always seemed to me to be like some kind of black magic and I have a bit hard time following, what's the root cause of this issue.

**Comment by lachlan-00** [comment]
nextcloud is stripping the .php from the url?

You could probably just set a new location in the config for the server
```
    location /apps/music/ampache/server/ {
    }
```

and then remove the fastcgi_split_path_info or copy the default settings without it into the folder

(DISCLAIMER that i have no idea how to use nginx)

**Comment by pled** [comment]
@paulijar : yes, I agree this is related to nginx configuration and not Music App. 

And looking at that [nextcloud documentation page](https://docs.nextcloud.com/server/stable/admin_manual/installation/nginx.html#nginx-config), it is mentionned that only Apache is officially supported. 
Nginx configuration examples are community-maintained...

Nevertheless, applying the `try_files` directive suggested by @aleixq makes Power Ampache able to connect to Nextcloud Music app. 
So may be the above documentation could add that issue in the [Tip and tricks](https://docs.nextcloud.com/server/stable/admin_manual/installation/nginx.html#tips-and-tricks) section ? Not sure who and how to contact someone about this... I am sure nginx is widely used, specially with docker Nextcloud implementations.

**Comment by paulijar** [comment]
I suspect that https://github.com/nextcloud/documentation/issues would be the correct place to request changes/additions to the Nextcloud documentation.

**Comment by paulijar** [comment]
I'll close this issue now as there's no changes suggested for the Music app itself.
